### PR TITLE
Cleanup Consul integration code

### LINKED ISSUE #1118: Eliminate dependency from proxy on platform
Re: https://github.com/istio/pilot/commit/c54e6ec0a5cab3bce8b67af05b67d464422bcf65
We require that proxy does not depend directly on the platform. All interactions should go through the model abstraction, to decoupling the behavior from the environment. 
I'm not sure why the PR got merged as-is, please re-work it to avoid bundling platform with the proxy.

### PR DESCRIPTION
**Pull request by GregHanson**
**What this PR does / why we need it**:
Follow up to premature merge of PR #1000 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
Resolves #1118 

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by rshriram** [comment]
Need more explanation please. From what I gathered, zijun's PR was non
functional. What exactly is done in this PR?


-- 

~shriram

File: proxy/envoy/net.go [hunk 1]
```diff
@@ -95,17 +70,3 @@ func getPrivateIPIfAvailable() net.IP {
...
-	for _, address := range addrs {
-		// check the address type and if it is not a loopback the display it
-		if ipnet, ok := address.(*net.IPNet); ok && !ipnet.IP.IsLoopback() {
-			if ipnet.IP.To4() != nil {
-				return ipnet.IP
-			}
-		}
-	}
-	return net.IPv4zero
-}
```
Comment (rshriram) [inline proxy/envoy/net.go]: Where is the replacement? You removed these and you are still using these functions in main

Comment (kyessenov) [reply proxy/envoy/net.go]: I'm not sure using local inet is a good idea. How do we pick the right interface if there are multiple available? What about ipv6?

Comment (rshriram) [reply proxy/envoy/net.go]: He is removing this function.. See this: https://github.com/istio/pilot/pull/1120/files#diff-10b44b0c268f7471a1cd69e4e0f59cc0L80 which is more generic.

Comment (rshriram) [reply proxy/envoy/net.go]: @GregHanson I feel that this file is not envoy specific. We could probably move this into proxy/

**Pull request merged**